kernel/src/cpu/percpu.rs

```
@@ -755,11 +749,12 @@ impl PerCpu {
         let vmsa = vmsa_ref.vmsa();
         self.apic
             .borrow_mut()
+            .unwrap()
```

This can panic here now, should we return an error instead?

---

Oh, I was reviewing this PR per-commit, so I noticed now you fixed in the next commit.
Not a strong opinion, but I'd introduce the error/option before this commit.

---

Based solely on the first commit, this code should be unreachable when `self.apic` is `None` because the caller will check whether APIC emulation is enabled.  In such a case, this call is technically illegal, in which case panicking is appropriate.  Changing the error paths is an improvement, but even if it were not done, the panic here is reasonable.